component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandler.java

```diff
@@ -45,128 +41,96 @@
                     String.format(ERROR_CODE_ERROR_ACCESS_USER_REALM.getMessage(), username, e));
-        } catch (CryptoException e) {
-            throw new BackupCodeException(ERROR_CODE_ERROR_ENCRYPT_BACKUP_CODE.getCode(),
-                    ERROR_CODE_ERROR_ENCRYPT_BACKUP_CODE.getMessage(), e);
         }
-        return claims;
+        return generatedBackupCodes;
     }
 
     /**
      * Update backup code claims and return backup codes for the user.
      *
-     * @param claims   Map with the backup code claims.
+     * @param generatedBackupCodes Generated backup codes as a comma separated string.
      * @param username Username of the user.
      * @return Backup codes for the user.
      * @throws BackupCodeException If an error occurred while updating backup codes.
      */
-    public static List<String> updateBackupCodes(Map<String, String> claims, String username)
+    public static List<String> updateBackupCodes(List<String> generatedBackupCodes, String username)
             throws BackupCodeException {
 
         String tenantAwareUsername;
-        String encryptedBackupCodes = claims.get(BACKUP_CODES_CLAIM);
+        ArrayList<String> hashedBackupCodesList = new ArrayList<>();
+        Map<String, String> claims = new HashMap<>();
+
         try {
             UserRealm userRealm = BackupCodeUtil.getUserRealm(username);
             if (userRealm != null) {
+                for (String backupCode : generatedBackupCodes) {
+                    hashedBackupCodesList.add(BackupCodeUtil.generateHashString(backupCode));
+                }
+                claims.put(BACKUP_CODES_CLAIM, String.join(",", hashedBackupCodesList));
+                claims.put(BACKUP_CODES_ENABLED_CLAIM, "true");
                 tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
                 userRealm.getUserStoreManager().setUserClaimValues(tenantAwareUsername, claims, null);
             }
         } catch (UserStoreException e) {
             throw new BackupCodeException(ERROR_CODE_ERROR_ACCESS_USER_REALM.getCode(),
                     String.format(ERROR_CODE_ERROR_ACCESS_USER_REALM.getMessage(), username, e));
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
         }
-        String backupCodes = null;
-        try {
-            if (StringUtils.isNotEmpty(encryptedBackupCodes)) {
-                backupCodes = BackupCodeUtil.decrypt(encryptedBackupCodes);
-            }
-        } catch (CryptoException e) {
-            throw new BackupCodeException(ERROR_CODE_ERROR_DECRYPT_BACKUP_CODE.getCode(),
-                    ERROR_CODE_ERROR_DECRYPT_BACKUP_CODE.getMessage(), e);
-        }
-        if (StringUtils.isEmpty(backupCodes)) {
-            return Collections.emptyList();
-        } else {
-            return new ArrayList<>(new ArrayList<>(Arrays.asList(backupCodes.split(","))));
-        }
+
```

Comment:
Let's remove the extra newlines

Reply:
addressed with [7d9674b](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/7d9674b6bf172b43c30479b0d64fdfa6f8130373)